src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  static aliveAnimals() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+
+    return Animal.alive;
+  }
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(victim) {
+    if (victim instanceof Herbivore && !victim.hidden) {
+      victim.health -= 50;
+      Animal.aliveAnimals();
+    }
```

Comment:
The `bite` method implementation is correct as it checks if the victim is an instance of Herbivore and not hidden before applying damage. Also, it updates the `Animal.alive` array by calling `Animal.aliveAnimals`, which is a good practice to ensure the array is always up-to-date.